caput/memh5.py

```diff
@@ -2103,11 +2115,22 @@ def index_map(self):
             Entries are 1D arrays used to interpret the axes of datasets.
 
         """
+        return ro_dict({k: v[:] for k, v in self._data["index_map"].items()})
 
-        out = {}
-        for name, value in self._data["index_map"].items():
-            out[name] = value[:]
-        return ro_dict(out)
+    @property
+    def index_attrs(self):
+        """Exposes the attributes of each index_map entry.
+
+        Allows the user to implement custom behaviour associated with
+        the axis. Assignment to this dictionary does nothing but it does
+        allow attribute values to be changed.
+
+        Returns
+        -------
+        index_attrs : read-write dict
+            Attribute dicts for each index_map entry
+        """
+        return rw_dict({k: v.attrs for k, v in self._data["index_map"].items()})
```

Comment:
Hmm, this might actually be left over from some development and not needed anymore. At one point, trying to modify the individual `attr` dicts would trigger an error related to `ro_dict` not have a `__setitem__` method, but I don't see why that would happen anymore so it can probably just be a `ro_dict`

Reply:
Yup, seems like the `rw_dict` is no longer needed